flow/record/fieldtypes/__init__.py

```diff
@@ -547,23 +547,63 @@ def _unpack(cls, data):
         return data
 
 
+def _is_posixlike_path(path: Any):
+    return isinstance(path, pathlib.PurePath) and "\\" not in (path._flavour.sep, path._flavour.altsep)
+
+
+def _is_windowslike_path(path: Any):
+    return isinstance(path, pathlib.PurePath) and "\\" in (path._flavour.sep, path._flavour.altsep)
+
+
 class path(pathlib.PurePath, FieldType):
     def __new__(cls, *args):
+        # This is modelled after pathlib.PurePath's __new__(), which means you
+        # will never get an instance of path, only instances of either
+        # posix_path or windows_path.
         if cls is path:
+            # By default, path will behave differently on windows and posix
+            # systems, similarly as pathlib.PurePath does.
             cls = windows_path if os.name == "nt" else posix_path
-            if len(args) > 0:
-                path_ = args[0]
-                if isinstance(path_, pathlib.PureWindowsPath):
+
+            # Try to determine the path behaviour based on the first path part
+            # in args that has pathlib.PurePath traits.
+            for path_part in args:
+                if isinstance(path_part, pathlib.PureWindowsPath):
+                    cls = windows_path
+                    # The (string) representation of a pathlib.PureWindowsPath is not round trip equivalent if a
+                    # path starts with a \ or / followed by a drive letter, e.g.: \C:\...
+                    # Meaning:
+                    #
+                    # str(PureWindowsPath(r"\C:\WINDOWS/Temp")) !=
+                    # str(PureWindowsPath(PureWindowsPath(r"\C:\WINDOWS/Temp"))),
+                    #
+                    # repr(PureWindowsPath(r"\C:\WINDOWS/Temp")) !=
+                    # repr(PureWindowsPath(PureWindowsPath(r"\C:\WINDOWS/Temp"))),
+                    #
+                    # This would be the case though when using PurePosixPath instead.
+                    #
+                    # This construction works around that by converting all path parts that are
+                    # pathlib.PureWindowsPath descendants to strings first.
+                    args = (str(arg) if isinstance(arg, pathlib.PureWindowsPath) else arg for arg in args)
```

Comment:
After determining the first path type, does converting all args to `str` work instead of specifically checking if it's a different type?

Reply:
Yes that would work. Technically it would work when if only the very first part is a `PureWindowsPath` only that part is converted to string.

I guess the idea is to make things a bit faster? I'll check the speed differences between casting everything to string vs converting args to a list and only converting the first part.

Reply:
I ran some tests and as expected, the option to always convert all parts if the first _encountered_ `PurePath` subclass is a `WindowsPurePath` is more or less constant time.

If we only convert the very _first_ part if it is a `WindowsPurePath` there is a slight slow down (due to an `enumerate` and extra `if` statement. If however The first part is not a `WindowsPurePath` it is slightly faster.

```
all_to_string ; path parts: 2 ; time: 0.5000176659996214
all_to_string ; path parts: 5 ; time: 0.4989467599989439
all_to_string ; path parts: 10 ; time: 0.5002165209989471
all_to_string ; path parts: 3 ; time: 0.5717005929982406
all_to_string ; path parts: 6 ; time: 0.5740014379989589
all_to_string ; path parts: 11 ; time: 0.572631115999684

first_to_string ; path parts: 2 ; time: 0.6870770460009226
first_to_string ; path parts: 5 ; time: 0.6977602070001012
first_to_string ; path parts: 10 ; time: 0.7180876769998576
first_to_string ; path parts: 3 ; time: 0.43469748999996227
first_to_string ; path parts: 6 ; time: 0.42878352199841174
first_to_string ; path parts: 11 ; time: 0.43002194500149926
```

Here the 2, 5 and 10 path parts have a `WindowsPurePath` as first part and the 3, 6 and 11 a string.

Reply:
I would prefer the `all_string` for consistency and predictability in this case.